plugins/janus_videocall.c

```diff
@@ -1358,20 +1358,15 @@ static void *janus_videocall_handler(void *data) {
 			json_t *msg_simulcast = json_object_get(msg->jsep, "simulcast");
 			if(msg_simulcast && janus_get_codec_pt(msg_sdp, "vp8") > 0) {
 				JANUS_LOG(LOG_VERB, "VideoCall callee (%s) is going to do simulcasting\n", session->username);
-				session->ssrc[0] = json_integer_value(json_object_get(msg_simulcast, "ssrc-0"));
-				session->ssrc[1] = json_integer_value(json_object_get(msg_simulcast, "ssrc-1"));
-				session->ssrc[2] = json_integer_value(json_object_get(msg_simulcast, "ssrc-2"));
+		        int rid_ext_id = -1;
+				janus_rtp_simulcasting_prepare(msg_simulcast, &rid_ext_id, session->ssrc, session->rid);
+				session->sim_context.rid_ext_id = rid_ext_id;
 			} else {
 				int i=0;
 				for(i=0; i<3; i++) {
 					session->ssrc[i] = 0;
 					g_free(session->rid[0]);
 					session->rid[0] = NULL;
-					if(peer) {
-						peer->ssrc[i] = 0;
-						g_free(peer->rid[0]);
-						peer->rid[0] = NULL;
-					}
```

Comment:
This is the only fix that I think does belong here: in fact, it was incorrectly disabling the previously configured simulcast settings for the peer (the caller in this instance).

Reply:
yes